src/Service/DynamoDb/src/Input/CreateTableInput.php

```diff
@@ -0,0 +1,426 @@
+     *
+     * @see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DataModel.html
+     * @required
+     *
+     * @var KeySchemaElement[]
+     */
+    private $KeySchema;
+
+    /**
+     * One or more local secondary indexes (the maximum is 5) to be created on the table. Each index is scoped to a given
+     * partition key value. There is a 10 GB size limit per partition key value; otherwise, the size of a local secondary
+     * index is unconstrained.
+     *
+     * @var LocalSecondaryIndex[]
+     */
+    private $LocalSecondaryIndexes;
+
+    /**
+     * One or more global secondary indexes (the maximum is 20) to be created on the table. Each global secondary index in
+     * the array includes the following:.
+     *
+     * @var GlobalSecondaryIndex[]
+     */
+    private $GlobalSecondaryIndexes;
+
+    /**
+     * Controls how you are charged for read and write throughput and how you manage capacity. This setting can be changed
+     * later.
+     *
+     * @var BillingMode::*|null
+     */
+    private $BillingMode;
+
+    /**
+     * Represents the provisioned throughput settings for a specified table or index. The settings can be modified using the
+     * `UpdateTable` operation.
+     *
+     * @var ProvisionedThroughput|null
+     */
+    private $ProvisionedThroughput;
+
+    /**
+     * The settings for DynamoDB Streams on the table. These settings consist of:.
+     *
+     * @var StreamSpecification|null
+     */
+    private $StreamSpecification;
+
+    /**
+     * Represents the settings used to enable server-side encryption.
+     *
+     * @var SSESpecification|null
+     */
+    private $SSESpecification;
+
+    /**
+     * A list of key-value pairs to label the table. For more information, see Tagging for DynamoDB.
+     *
+     * @see https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Tagging.html
+     *
+     * @var Tag[]
+     */
+    private $Tags;
+
+    /**
+     * @param array{
+     *   AttributeDefinitions?: \AsyncAws\DynamoDb\ValueObject\AttributeDefinition[],
+     *   TableName?: string,
+     *   KeySchema?: \AsyncAws\DynamoDb\ValueObject\KeySchemaElement[],
+     *   LocalSecondaryIndexes?: \AsyncAws\DynamoDb\ValueObject\LocalSecondaryIndex[],
+     *   GlobalSecondaryIndexes?: \AsyncAws\DynamoDb\ValueObject\GlobalSecondaryIndex[],
+     *   BillingMode?: \AsyncAws\DynamoDb\Enum\BillingMode::*,
+     *   ProvisionedThroughput?: \AsyncAws\DynamoDb\ValueObject\ProvisionedThroughput|array,
+     *   StreamSpecification?: \AsyncAws\DynamoDb\ValueObject\StreamSpecification|array,
+     *   SSESpecification?: \AsyncAws\DynamoDb\ValueObject\SSESpecification|array,
+     *   Tags?: \AsyncAws\DynamoDb\ValueObject\Tag[],
+     * } $input
+     */
+    public function __construct(array $input = [])
+    {
+        $this->AttributeDefinitions = array_map(function ($item) { return AttributeDefinition::create($item); }, $input['AttributeDefinitions'] ?? []);
```

Comment:
Isn't it faster ?:
```diff
- array_map(function ($item) { return AttributeDefinition::create($item); }, $input['AttributeDefinitions'] ?? []);
+ array_map([AttributeDefinition::class, 'create'], $input['AttributeDefinitions'] ?? []);
```

Reply:
Why, isnt it the exact same thing but different syntax?

Reply:
yes, it's equivalent, just a small performance optimization.
By wraping the Factory in a closure, we performs 2 methods call, when 1 is enougth.